packages/teleterm/src/services/tshd/middleware.ts

```diff
@@ -91,14 +93,41 @@ export const withLogging = (logger: Logger) => {
         });
       },
 
-      sendMessage(message: any, next: Function) {
-        logger.info(`send: ${method}(${JSON.stringify(message.toObject())})`);
+      sendMessage(message, next) {
+        logger.info(
+          `send: ${method}(${JSON.stringify(
+            filterPasswords(message.toObject())
+          )})`
+        );
         next(message);
       },
     };
 
     return new grpc.InterceptingCall(nextCall(options), params);
   };
-
-  return interceptor;
 };
+
+function filterSensitiveProperties(
+  toFilter: object,
+  filteredProperties: string[]
+): object {
+  return transform(
+    toFilter,
+    (result: object, value: any, key: any) => {
+      if (
+        filteredProperties.some(
+          filteredProp => typeof key === 'string' && key.includes(filteredProp)
+        )
+      ) {
+        result[key] = '~FILTERED~';
+      } else {
+        if (isObject(value)) {
```

Comment:
Small nit, but I think this could made into `else if` instead which would save one nesting level, thus also making the "default"/"else" case easier to notice.

Reply:
Done by using early returns, IMO it is even easier to read than `else if`.